src/audio/codec_adapter/codec/waves.c

```diff
@@ -0,0 +1,744 @@
+
+	status = MaxxEffect_Initialize(waves_codec->effect, i_formats, 1, o_formats, 1);
+
+	if (status) {
+		comp_err(dev, "waves_effect_init() MaxxEffect_Initialize returned %d", status);
+		return -EINVAL;
+	}
+
+	comp_dbg(dev, "waves_effect_init() done");
+	return 0;
+}
+
+/* allocate additional buffers for MaxxEffect */
+static int waves_effect_buffers(struct comp_dev *dev)
+{
+	struct codec_data *codec = comp_get_codec(dev);
+	struct waves_codec_data *waves_codec = codec->private;
+	MaxxStatus_t status;
+	int ret;
+	void *i_buffer = NULL, *o_buffer = NULL, *response = NULL;
+
+	comp_dbg(dev, "waves_effect_buffers() start");
+
+	status = MaxxEffect_GetMessageMaxSize(waves_codec->effect, &waves_codec->request_max_bytes,
+					      &waves_codec->response_max_bytes);
+
+	if (status) {
+		comp_err(dev, "waves_effect_buffers() MaxxEffect_GetMessageMaxSize returned %d",
+			 status);
+		ret = -EINVAL;
+		goto err;
+	}
+
+	response = codec_allocate_memory(dev, waves_codec->response_max_bytes, 16);
+	if (!response) {
+		comp_err(dev, "waves_effect_buffers() failed to allocate %d bytes for response",
+			 waves_codec->response_max_bytes);
+		ret = -ENOMEM;
+		goto err;
+	}
+
+	i_buffer = codec_allocate_memory(dev, waves_codec->buffer_bytes, 16);
+	if (!i_buffer) {
+		comp_err(dev, "waves_effect_buffers() failed to allocate %d bytes for i_buffer",
+			 waves_codec->buffer_bytes);
+		ret = -ENOMEM;
+		goto err;
+	}
+
+	o_buffer = codec_allocate_memory(dev, waves_codec->buffer_bytes, 16);
+	if (!o_buffer) {
+		comp_err(dev, "waves_effect_buffers() failed to allocate %d bytes for o_buffer",
+			 waves_codec->buffer_bytes);
+		ret = -ENOMEM;
+		goto err;
+	}
+
+	waves_codec->i_buffer = i_buffer;
+	waves_codec->o_buffer = o_buffer;
+	waves_codec->response = response;
+	codec->cpd.in_buff = waves_codec->i_buffer;
+	codec->cpd.in_buff_size = waves_codec->buffer_bytes;
+	codec->cpd.out_buff = waves_codec->o_buffer;
+	codec->cpd.out_buff_size = waves_codec->buffer_bytes;
+
+	comp_info(dev, "waves_effect_buffers() size response %d, i_buffer %d, o_buffer %d",
+		  waves_codec->response_max_bytes, waves_codec->buffer_bytes,
+		  waves_codec->buffer_bytes);
+
+	comp_dbg(dev, "waves_effect_buffers() done");
+	return 0;
+
+err:
+	if (i_buffer)
+		codec_free_memory(dev, i_buffer);
+	if (o_buffer)
+		codec_free_memory(dev, o_buffer);
+	if (response)
+		codec_free_memory(dev, response);
+	return ret;
+}
+
+/* get MaxxEffect revision */
+static int waves_effect_revision(struct comp_dev *dev)
+{
+	struct codec_data *codec = comp_get_codec(dev);
+	struct waves_codec_data *waves_codec = codec->private;
+	const char *revision = NULL;
+	uint32_t revision_len;
+	MaxxStatus_t status;
+
+	comp_info(dev, "waves_effect_revision() start");
+
+	status = MaxxEffect_Revision_Get(waves_codec->effect, &revision, &revision_len);
+
+	if (status) {
+		comp_err(dev, "waves_effect_revision() MaxxEffect_Revision_Get returned %d",
+			 status);
+		return -EINVAL;
+	}
+
+#if CONFIG_TRACE
```

Comment:
almost wonder if this should be behind VERBOSE_TRACE

Reply:
agreed, done